src/auth_client_demo_assets/svelte/src/stores/auth.js

```diff
@@ -0,0 +1,131 @@
+import { AuthClient } from "@dfinity/auth-client";
+import { readable } from "svelte/store";
+import { createActor, canisterId } from "../../../../declarations/whoami";
+
+const defaultOptions = {
+  /**
+   *  @type {import("@dfinity/auth-client").AuthClientCreateOptions}
+   */
+  createOptions: {
+    idleOptions: {
+      // Set to true if you do not want idle functionality
+      disableIdle: true,
+    },
+  },
+  /**
+   * @type {import("@dfinity/auth-client").AuthClientLoginOptions}
+   */
+  loginOptions: {
+    identityProvider:
+      process.env.DFX_NETWORK === "ic"
+        ? "https://identity.ic0.app/#authorize"
+        : `http://localhost:4943?canisterId=${process.env.CANISTER_ID_INTERNET_IDENTITY}#authorize`,
+  },
+};
+
+/**
+ *
+ * @param {import("@dfinity/agent").Identity} identity
+ * @returns {import("@dfinity/agent").ActorSubclass<import("../../../../declarations/whoami/whoami.did")._SERVICE>}
+ */
+function actorFromIdentity(identity) {
+  return createActor(canisterId, {
+    agentOptions: {
+      host:
+        process.env.DFX_NETWORK === "ic"
```

Comment:
why not using vite environment variable (`import.env...`)?

Reply:
mainly because I'm relying on the `.env` file from `dfx`

Reply:
Gotcha